pwiz_tools/Skyline/Model/Lib/BiblioSpecLite.cs

```diff
@@ -1156,7 +1155,7 @@ private bool Load(ILoadMonitor loader, IProgressStatus status, bool cached, out
             }
             catch (Exception x)
             {
-                failureException = new Exception(FormatErrorMessage(x), x);
+                var failureException = new Exception(FormatErrorMessage(x), x);

```

Comment:
This previous "catch (InvalidDataException x)" and "catch (IOException x)" can be combined with this "catch (Exception x)" since each catch block is doing the same thing.

However, since we want some things in the "catch (Exception)" block to be reported to Exception web, what we now recommend doing is using the new "ExceptionUtil.IsProgrammingDefect" method to check whether the error should be reported to exception web:
```
catch (Exception x) 
{
    if (!cached) 
    {
        if (ExceptionUtil.IsProgrammingDefect(x)) 
        {
            throw;
        }
        if (!cached)
        {
            loader.UpdateProgress(status.ChangeErrorException(new Exception(FormatErrorMessage(x), x)));
        }
    }
}
```

Reply:
I suspect that ReSharper will complain about the second if (!cached) always being true. Otherwise, I agree that there could be catch (Exception e) clause and I like the new IsProgrammingDefect() check.

Reply:
> This previous "catch (InvalidDataException x)" and "catch (IOException x)" can be combined with this "catch (Exception x)" since each catch block is doing the same thing.

I was assuming that there was value in preserving the exception Type where it was known - you don't think so?

Reply:
I think it is enough to preserve that, and far more importantly the original stack trace, as you have by making that exception the inner exception to the one you create.

Reply:
You no longer actually throw the exception, and you are clear that the only thing left to throw to is the unhandled exception handler, which we still want for programming error type exceptions.

Makes me realize that you may still not quite want what Nick suggested? Is there anything to be gained with FormatErrorMessage() when it was a programming error? i.e.

if (ExceptionUtil.IsProgrammingDefect(x)) 
{
    throw new Exception(FormatErrorMessage(x), x);
}

Reply:
>  Is there anything to be gained with FormatErrorMessage() when it was a programming error? 

Yes, it includes potentially useful information about the library filename and contents. I think it should be used in this instance.